File: src/Clueless/HallwaySpace.m
//
//  HallwaySpace.m
//  Clueless
//
//  Created by John Gilday on 11/23/12.
//  Copyright (c) 2012 Johns Hopkins University. All rights reserved.
//

#import "HallwaySpace.h"

@implementation HallwaySpace

- (HallwaySpace*) initWithRoomsToConnect:(Room *)room1 ConnectsTo:(Room *)room2
{
    self = [super init];
    
    // Create ID from room IDs
    NSMutableString *idFromRooms = [[NSMutableString alloc] initWithString:room1.spaceId];
    [idFromRooms appendString:@"-"];
    [idFromRooms appendString:room2.spaceId];
    self.spaceId = idFromRooms;
    
    // Set up navigation links
    [self connectSpace:room1];
    [self connectSpace:room2];
    
    return self;
}

@end

File: src/clueless-android/src/edu/jhu/ep/butlerdidit/domain/HallwaySpace.java
package edu.jhu.ep.butlerdidit.domain;

public class HallwaySpace extends GameBoardSpace {
	
	public static String hallwayIdFromRooms(Room room1, Room room2) {
		return String.format("%s-%s", room1.getSpaceId(), room2.getSpaceId());
	}

    public HallwaySpace(Room room, Room connectsTo) {
        super(HallwaySpace.hallwayIdFromRooms(room, connectsTo));

        connectSpace(room);
        connectSpace(connectsTo);
    }
}
